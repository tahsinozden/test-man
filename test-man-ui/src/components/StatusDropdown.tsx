import React, {useEffect} from "react";
import {Dropdown, DropdownButton} from "react-bootstrap";

interface StatusDropDownProps {
    currentStatus: string;
    onStatusSelect: (newStatus: string) => void;
}

const StatusDropdown: React.FC<StatusDropDownProps> = ({currentStatus, onStatusSelect}: StatusDropDownProps) => {
    const statuses = ["Passed", "Failed"];
    const statusDropdownItems: any = [];

    useEffect(() => {
        for (let status of statuses) {
            statusDropdownItems.push(
                <Dropdown.Item as="label" eventKey={status} key={status}>{status}</Dropdown.Item>
            );
        }
    }, []);

    return (
        <>
            {/*FIXME: based on the status string, the size changes*/}
            <DropdownButton
                id="dropdown-item-button"
                title={currentStatus}
                variant="secondary"
                onSelect={(selectedValue) => {
                    if (selectedValue) {
                        onStatusSelect(selectedValue);
                    }
                }}
            >
                {statusDropdownItems}
            </DropdownButton>
        </>
    );
}

export default StatusDropdown;