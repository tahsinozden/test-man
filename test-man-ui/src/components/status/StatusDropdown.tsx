import React, {useEffect, useState} from "react";
import TestStatus from "../../models/TestStatus";
import testManagerApi from "../../api/TestManagerApi";
import "./StatusDropDown.css"
import {Button, Dropdown} from "antd";
import {DownOutlined} from "@ant-design/icons";
import StatusMenu from "./StatusMenu";

interface StatusDropDownProps {
    currentStatus: string;
    onStatusSelect: (newStatus: string) => void;
}

const StatusDropdown: React.FC<StatusDropDownProps> = ({currentStatus, onStatusSelect}: StatusDropDownProps) => {
    const [statuses, setStatuses] = useState<string[]>([]);
    const [selected, setSelected] = useState(TestStatus.Undefined.toString());
    const colorByStatus = new Map([
            [TestStatus.Undefined.toString(), "grey"],
            [TestStatus.Passed.toString(), "green"],
            [TestStatus.Failed.toString(), "red"]
        ]
    );

    useEffect(() => {
        testManagerApi.getAllStatuses().then(data => setStatuses(data));
    }, []);

    useEffect(() => {
        setSelected(currentStatus);
    }, [currentStatus]);

    const handleMenuClick = (e: any) => {
        onStatusSelect(e.key);
        setSelected(e.key);
    }

    return (
        <>
            <Dropdown overlay={<StatusMenu statuses={statuses} onClick={handleMenuClick}/>}>
                <Button data-testid="current-status-button" type="primary" className="status-button"
                        style={{background: colorByStatus.get(selected)}}>
                    {selected} <DownOutlined/>
                </Button>
            </Dropdown>
        </>
    );
}

export default StatusDropdown;