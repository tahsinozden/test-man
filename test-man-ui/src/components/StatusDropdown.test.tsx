import renderer from "react-test-renderer";
import StatusDropdown from "./StatusDropdown";
import "../jestGlobalMocks"

it('renders correctly when there is a current status', () => {
    const tree = renderer
        .create(<StatusDropdown currentStatus={"Passed"}
                                onStatusSelect={() => {
                                }}/>)
        .toJSON();
    expect(tree).toMatchSnapshot();
});

it('renders correctly when there is a empty status text', () => {
    const tree = renderer
        .create(<StatusDropdown currentStatus={""}
                                onStatusSelect={() => {
                                }}/>)
        .toJSON();
    expect(tree).toMatchSnapshot();
});

it('renders correctly when there is a current status "Passed"', () => {
    const tree = renderer
        .create(<StatusDropdown currentStatus={"Passed"}
                                onStatusSelect={() => {
                                }}/>)
        .toJSON();
    expect(tree).toMatchSnapshot();
});

it('renders correctly when there is a current status "Failed"', () => {
    const tree = renderer
        .create(<StatusDropdown currentStatus={"Failed"}
                                onStatusSelect={() => {
                                }}/>)
        .toJSON();
    expect(tree).toMatchSnapshot();
});

it('renders correctly when there is a current status "Undefined"', () => {
    const tree = renderer
        .create(<StatusDropdown currentStatus={"Undefined"}
                                onStatusSelect={() => {
                                }}/>)
        .toJSON();
    expect(tree).toMatchSnapshot();
});
