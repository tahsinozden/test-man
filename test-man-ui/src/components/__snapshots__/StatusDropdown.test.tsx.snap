// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly when there is a current status "Failed" 1`] = `
<button
  className="ant-btn ant-btn-primary ant-dropdown-trigger status-button"
  onClick={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  style={
    Object {
      "background": "red",
    }
  }
  type="button"
>
  <span>
    Failed 
  </span>
  <span
    aria-label="down"
    className="anticon anticon-down"
    role="img"
  >
    <svg
      aria-hidden="true"
      data-icon="down"
      fill="currentColor"
      focusable="false"
      height="1em"
      viewBox="64 64 896 896"
      width="1em"
    >
      <path
        d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
      />
    </svg>
  </span>
</button>
`;

exports[`renders correctly when there is a current status "Passed" 1`] = `
<button
  className="ant-btn ant-btn-primary ant-dropdown-trigger status-button"
  onClick={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  style={
    Object {
      "background": "green",
    }
  }
  type="button"
>
  <span>
    Passed 
  </span>
  <span
    aria-label="down"
    className="anticon anticon-down"
    role="img"
  >
    <svg
      aria-hidden="true"
      data-icon="down"
      fill="currentColor"
      focusable="false"
      height="1em"
      viewBox="64 64 896 896"
      width="1em"
    >
      <path
        d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
      />
    </svg>
  </span>
</button>
`;

exports[`renders correctly when there is a current status "Undefined" 1`] = `
<button
  className="ant-btn ant-btn-primary ant-dropdown-trigger status-button"
  onClick={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  style={
    Object {
      "background": "grey",
    }
  }
  type="button"
>
  <span>
    Undefined 
  </span>
  <span
    aria-label="down"
    className="anticon anticon-down"
    role="img"
  >
    <svg
      aria-hidden="true"
      data-icon="down"
      fill="currentColor"
      focusable="false"
      height="1em"
      viewBox="64 64 896 896"
      width="1em"
    >
      <path
        d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
      />
    </svg>
  </span>
</button>
`;

exports[`renders correctly when there is a current status 1`] = `
<button
  className="ant-btn ant-btn-primary ant-dropdown-trigger status-button"
  onClick={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  style={
    Object {
      "background": "green",
    }
  }
  type="button"
>
  <span>
    Passed 
  </span>
  <span
    aria-label="down"
    className="anticon anticon-down"
    role="img"
  >
    <svg
      aria-hidden="true"
      data-icon="down"
      fill="currentColor"
      focusable="false"
      height="1em"
      viewBox="64 64 896 896"
      width="1em"
    >
      <path
        d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
      />
    </svg>
  </span>
</button>
`;

exports[`renders correctly when there is a empty status text 1`] = `
<button
  className="ant-btn ant-btn-primary ant-dropdown-trigger status-button"
  onClick={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  style={
    Object {
      "background": undefined,
    }
  }
  type="button"
>
  <span>
     
  </span>
  <span
    aria-label="down"
    className="anticon anticon-down"
    role="img"
  >
    <svg
      aria-hidden="true"
      data-icon="down"
      fill="currentColor"
      focusable="false"
      height="1em"
      viewBox="64 64 896 896"
      width="1em"
    >
      <path
        d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
      />
    </svg>
  </span>
</button>
`;
